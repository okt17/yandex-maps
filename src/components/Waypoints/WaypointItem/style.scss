.app__waypoints__item {
  background-color: #64B5F6;
  border: 1px solid #1565C0;
  border-radius: 5px;
  box-shadow: 0 0 10px rgba(0,0,0,0.5);
  cursor: -webkit-grab;
  cursor: -moz-grab;
  display: flex;
  padding: 0.5rem;

  &:hover {
    background-color: #42A5F5;
  }

  &__button {
    color: white !important;
  }

  &__container {
    flex: 5 5 100%;
    padding: 0.5rem;
  }
}
