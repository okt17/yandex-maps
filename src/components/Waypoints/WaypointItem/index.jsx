import React from 'react';
import PropTypes from 'prop-types';
import { IconButton, Icon } from '@material-ui/core';
import './style';

const BCN = 'app__waypoints__item'; // Base Class Name

const toFixed = n => n.toFixed(3);

class WaypointItem extends React.PureComponent {
  static propTypes = {
    deleteWaypoint: PropTypes.func.isRequired,
    index: PropTypes.number.isRequired,
    waypoint: PropTypes.object.isRequired,
  };
  handleDeleteButtonClick = () => this.props.deleteWaypoint(this.props.waypoint.id);
  handleDragStart = (event) => {
    event.dataTransfer.effectAllowed = 'move';
    event.dataTransfer.setData('waypoint', this.props.index);

    return true;
  };
  render() {
    const {
      waypoint: {
        name,
        coordinates,
      },
    } = this.props;

    return (
      <div
        className={BCN}
        draggable
        onDragStart={this.handleDragStart}
      >
        <span className={`${BCN}__container`}>
          <div>{name}</div>

          <div>{coordinates.map(toFixed).join(', ')}</div>
        </span>

        <IconButton
          color='secondary'
          onClick={this.handleDeleteButtonClick}
          size='small'
          title='Удалить точку маршрута'
        >
          <Icon>clear</Icon>
        </IconButton>
      </div>
    );
  }
}

export default WaypointItem;
