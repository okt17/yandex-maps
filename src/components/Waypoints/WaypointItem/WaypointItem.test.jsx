/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import WaypointItem from '.';

test('WaypointItem component test', () => {
  const component = renderer.create(<WaypointItem
    deleteWaypoint={() => {}}
    index={1337}
    waypoint={{
      name: 'waypoint',
      coordinates: [37, 13],
    }}
  />);

  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
