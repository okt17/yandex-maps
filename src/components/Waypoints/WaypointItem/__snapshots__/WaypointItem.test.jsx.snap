// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WaypointItem component test 1`] = `
<div
  className="app__waypoints__item"
  draggable={true}
  onDragStart={[Function]}
>
  <span
    className="app__waypoints__item__container"
  >
    <div>
      waypoint
    </div>
    <div>
      37.000, 13.000
    </div>
  </span>
  <button
    className="MuiButtonBase-root-7 MuiIconButton-root-1 MuiIconButton-colorSecondary-4"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    size="small"
    tabIndex="0"
    title="Удалить точку маршрута"
    type="button"
  >
    <span
      className="MuiIconButton-label-6"
    >
      <span
        aria-hidden="true"
        className="material-icons MuiIcon-root-10"
      >
        clear
      </span>
    </span>
    <span
      className="MuiTouchRipple-root-16"
    />
  </button>
</div>
`;
